blocksync/buffer.go

```diff
@@ -66,6 +67,11 @@ func (b *blockBuffer) Flush(blk *blockchain.Block) (bool, bCheckinResult) {
 		}
 		delete(b.blocks, heightToSync)
 		if err := commitBlock(b.bc, b.ap, blk); err != nil {
+			if err == db.ErrAlreadyExist {
```

Comment:
Sorry, I just double check , and i think the right logic here is
```suggestion
			if err == db.ErrAlreadyExist {
				l.Info().Uint64("syncHeight", heightToSync).Msg("Block already exists.")
			} else {
				l.Error().Err(err).Uint64("syncHeight", heightToSync).Msg("Failed to commit the block.")
				break
			}
```

And then, we don't need L75-L81